scripts/cast-on-tenderly/index.js

```diff
@@ -19,33 +20,91 @@ if (!SPELL_ADDRESS) {
+                accept: 'application/json, text/plain, */*',
+                'X-Access-Key': process.env.TENDERLY_ACCESS_KEY,
+            },
+        });
+    } catch (error) {
+        console.error('makeTenderlyApiRequest error', error.response);
+        throw new Error(`tenderly request failed with: ${error.code}`);
+    }
 };
 
-const createTenderlyFork = async function () {
-    const response = await makeTenderlyApiRequest('/fork');
-    const forkId = response.data.simulation_fork.id;
-    const rpcUrl = `https://rpc.tenderly.co/fork/${forkId}`;
-    const forkUrl = `https://dashboard.tenderly.co/${process.env.TENDERLY_USER}/${process.env.TENDERLY_PROJECT}/fork/${forkId}`;
-    return { forkId, forkUrl, rpcUrl };
+const createTenderlyTestnet = async function (spellName) {
+    const slug = `${spellName.replace(/\s+/g, '-').toLowerCase()}-${randomUUID()}`;
+    const response = await makeTenderlyApiRequest({
+        method: 'post',
+        path: '/testnet/container',
+        body: {
+            slug,
+            displayName: spellName,
+            description: spellName,
+            networkConfig: {
+                networkId: NETWORK_ID,
+                blockNumber: 'latest',
+                baseFeePerGas: '1',
+                chainConfig: {
+                    chainId: NETWORK_ID,
+                },
+            },
+            explorerPage: 'DISABLED',
+            syncState: true,
+        },
+    });
+    const testnetId = response.data.container.id;
+    const rpcEndpointPublic = response.data.container.connectivityConfig.endpoints.find(
+        endpoint => endpoint.displayName === 'testnet'
+    );
+    const rpcEndpointPrivate = response.data.container.connectivityConfig.endpoints.find(
+        endpoint => endpoint.displayName === 'unlocked'
+    );
+    const explorerUrlPublic = `https://dashboard.tenderly.co/explorer/vnet/${rpcEndpointPublic.id}`;
+    return {
+        testnetId,
+        explorerUrlPublic,
+        rpcUrlPublic: rpcEndpointPublic.uri,
+        rpcUrlPrivate: rpcEndpointPrivate.uri,
+    };
 };
 
-const publishTenderlyTransaction = async function (forkId, transactionId) {
-    await makeTenderlyApiRequest(`/fork/${forkId}/transaction/${transactionId}/share`);
-    return `https://dashboard.tenderly.co/shared/fork/simulation/${transactionId}`;
+const publishTenderlyTestnet = async function (testnetId) {
+    const response = await makeTenderlyApiRequest({
+        method: 'put',
+        path: `/testnet/container/${testnetId}`,
+        body: {
+            explorerPage: 'ENABLED',
+        },
+    });
+    if (response.data?.container?.explorer_page !== 'ENABLED') {
+        throw new Error('failed to publish testnet');
+    }
 };
 
-const runSpell = async function () {
-    const { forkId, forkUrl, rpcUrl } = await createTenderlyFork();
-    console.info('private tenderly fork is created', forkUrl);
+const runSpell = async function (spellAddress) {
```

Comment:
Nit: I would name this function `castOnTenderly` so it's clear it is the main function of the script (the name is the same).
It's subjective, because one can easily check that by going to the very last line of the file.